### ISSUE TITLE
Is it possible to save checkpoints based on validation performance?

### ISSUE THREAD

**Issue by netanel-s** [tensorflow/models#5004]
### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code **: stock
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: CUDA 9
- **GPU model and memory**: 1080Ti, 12GB
- **Exact command to reproduce**: training and evaluating ssdlite_mobilenet_v2_coco_2018_05_09 (on COCO dataset) from checkpoint of mobilenet_v2_1.0_224 using object_detection/model_main.py.

### Describe the problem
Is it possible to save checkpoints based on validation performance?
I want to train my model, and decide which checkpoints to save based on the performance of the model on the validation set. I want to do so in order to save certain checkpoints (not only the latest ones) and maybe stop the training in some case when I cannot monitor the training.
I know that ValidationMonitor had a similar functionality, but it is deprecated, so I wonder if this functionality exists somewhere else which I couldn't find.
Thanks in advance.

**Comment by yinguobing** [comment]
It would be a lot better if this is possible.

**Comment by netanel-s** [comment]
@robieta , anything new regarding this issue? 
It's been 3 months now, any input will be appreciated.
Thanks.

**Comment by netanel-s** [comment]
@robieta , five months have passed, any comment would be appreciated.
Thanks.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by apacha** [comment]
The TF Object Detection API is dead (e.g., they specifically added the note, that TF 2.0 will not be supported). I recommend moving on to [mmdetection](https://github.com/open-mmlab/mmdetection). They offer the latest algorithms and it's quite easy to add callbacks for early stopping and saving checkpoints with PyTorch.
